connectors/src/connectors/confluence/index.ts

```diff
@@ -465,8 +469,33 @@ export class ConfluenceConnectorManager extends BaseConnectorManager<null> {
     return new Ok(contentNodes);
   }
 
+  private _cachedHierarchyMutex = new Mutex();
```

Comment:
I'm hesitant about adding `async-mutex` as it introduces an additional pattern. Since other connectors use Redis, perhaps we could align with them and adopt the same approach. WDYT?

Reply:
I agree and tried to do this via redis at first. From my understanding, it's not easy to do with redis because there is the same I/O concurrency issue:
- threads go check on redis if the key exist
- if yes, they grab the result, if not, they make the DB call and cache it -- but the DB call may take hundreds of ms, and only after that will the key in redis be updated
- 
=> we have the test and set issue requiring locking either way: multiple threads can read that the key is not set and all try to write it

Redis may have some locking primitives available, but then we'd actually need to introduce a new pattern too so in that case better to do it directly in JS